### PR TITLE
Use `MpModelWrapper` and `MpSerialExecutor`

### PR DESCRIPTION
**Pull request by jysohn23**
In our training colab notebooks. Fixes https://github.com/pytorch/xla/issues/2102

### CODE REVIEW

File: contrib/colab/mnist-training.ipynb [hunk 1]
```diff
@@ -205,28 +207,25 @@
...
+        "        download=True,\n",
+        "        transform=transforms.Compose(\n",
+        "            [transforms.ToTensor(), norm]))\n",
+        "    \n",
+        "    return train_dataset, test_dataset\n",
         "  \n",
-        "  if xm.is_master_ordinal():\n",
-        "    # Barrier: Master done downloading, other workers can proceed\n",
-        "    xm.rendezvous('download_only_once')\n",
+        "  # Only first run of get_dataset downloads dataset locally\n",
```
Comment (dlibenzi) [inline contrib/colab/mnist-training.ipynb:227]: I would explain a bit better.
Like:

Using the serial executor avoids multiple processes to download the same data.

File: contrib/colab/mnist-training.ipynb [hunk 2]
```diff
@@ -251,7 +250,9 @@
         "\n",
         "  # Get loss function, optimizer, and model\n",
         "  device = xm.xla_device()\n",
-        "  model = MNIST().to(device)\n",
+        "  # Only instantiate model weights once in memory\n",
+        "  wrapped_model = xmp.MpModelWrapper(MNIST())\n",
```
Comment (dlibenzi) [inline contrib/colab/mnist-training.ipynb:254]: The model wrapper works like the serial executor. You need to create it at global scope:

https://github.com/pytorch/xla/blob/bbb4932534027b1cddc5d65db39bb82486737cda/torch_xla/distributed/xla_multiprocessing.py#L309

File: contrib/colab/resnet18-training.ipynb [hunk 3]
```diff
@@ -262,40 +262,39 @@
...
+        "        train=False,\n",
+        "        download=True,\n",
+        "        transform=transform_test)\n",
+        "    \n",
+        "    return train_dataset, test_dataset\n",
         "  \n",
-        "  if xm.is_master_ordinal():\n",
-        "    # Barrier: Master done downloading, other workers can proceed\n",
-        "    xm.rendezvous('download_only_once')\n",
+        "  # Only first run of get_dataset downloads dataset locally\n",
```
Comment (dlibenzi) [inline contrib/colab/resnet18-training.ipynb:296]: Same comment apply here ...

File: contrib/colab/resnet18-training.ipynb [hunk 4]
```diff
@@ -320,7 +319,9 @@
         "\n",
         "  # Get loss function, optimizer, and model\n",
         "  device = xm.xla_device()\n",
-        "  model = ResNet18().to(device)\n",
+        "  # Only instantiate model weights once in memory\n",
+        "  wrapped_model = xmp.MpModelWrapper(ResNet18())\n",
```
Comment (dlibenzi) [inline contrib/colab/resnet18-training.ipynb:323]: And here.

**Comment by taylanbil** [comment]
Maybe it doesn't belong to this PR but we need to change the way we plot in this NB:

```python
          if rank == 0:
            plot_results(data.cpu(), pred.cpu(), target.cpu())
```

We're doing tensor work on one device only. We could plot on all cores and save to file from one core instead.

**Comment by dlibenzi** [comment]
> Maybe it doesn't belong to this PR but we need to change the way we plot in this NB:
> 
> ```python
>           if rank == 0:
>             plot_results(data.cpu(), pred.cpu(), target.cpu())
> ```
> 
> We're doing tensor work on one device only. We could plot on all cores and save to file from one core instead.

That's the perfect example for using the other new API:

https://github.com/pytorch/xla/blob/bbb4932534027b1cddc5d65db39bb82486737cda/torch_xla/core/xla_model.py#L678

File: contrib/colab/mnist-training.ipynb (hunk 1 above)
Comment (jysohn23) [reply contrib/colab/mnist-training.ipynb]: Done.

File: contrib/colab/mnist-training.ipynb (hunk 2 above)
Comment (jysohn23) [reply contrib/colab/mnist-training.ipynb]: Ah yes, thanks. Done.

File: contrib/colab/resnet18-training.ipynb (hunk 3 above)
Comment (jysohn23) [reply contrib/colab/resnet18-training.ipynb]: Done.

File: contrib/colab/resnet18-training.ipynb (hunk 4 above)
Comment (jysohn23) [reply contrib/colab/resnet18-training.ipynb]: Done.

**Comment by jysohn23** [comment]
> That's the perfect example for using the other new API:
> 
> https://github.com/pytorch/xla/blob/bbb4932534027b1cddc5d65db39bb82486737cda/torch_xla/core/xla_model.py#L678

Looked like there was a bug. Will send followup PR once https://github.com/pytorch/xla/pull/2114 is merge and we have a build with it.

File: contrib/colab/mnist-training.ipynb [hunk 5]
```diff
@@ -201,32 +203,32 @@
         "    x = self.fc2(x)\n",
         "    return F.log_softmax(x, dim=1)\n",
         "\n",
+        "# Only instantiate model weights once in memory\n",
```
Comment (dlibenzi) [inline contrib/colab/mnist-training.ipynb:206]: Let's end comments like English sentences, with a dot.
Same in the matching comment in the other file.

Comment (jysohn23) [reply contrib/colab/mnist-training.ipynb]: Done, now it's proper english 😄

**Pull request merged**